### ISSUE TITLE
Move admin to something else than /admin

### ISSUE THREAD

**Issue by Woseseltops** [Signbank/Global-signbank#832]
The Radboud Security Operations Center asks us to move the admin to another location than just `/admin`:

> Overweeg om de Django administrator login pagina te verbergen en niet default onder /admin te plaatsen.
Dit soort standaard URL’s zijn de eerste die gezocht zullen worden door hackers voor nader onderzoek en exploits.

While not hard to realize, it could be yet another source of confusion/frustration for new Signbank devs... other opinions wanted.

**Comment by vanlummelhuizen** [comment]
Perhaps one common replacement for `/admin` for all TSG projects?

Other option: shield `/admin` for ip-addresses outside the RU network.

**Comment by vanlummelhuizen** [comment]
> Perhaps one common replacement for `/admin` for all TSG projects?

We should be aware that hardcoding a alternative for `/admin` will still be public in the repo. Perhaps make it an instance specific setting.

Some interesting stuff: https://opensource.com/article/18/1/10-tips-making-django-admin-more-secure

**Comment by Woseseltops** [comment]
> We should be aware that hardcoding a alternative for /admin will still be public in the repo. Perhaps make it an instance specific setting.

I like this option! This way, the default will still be /admin so we won't confuse new devs trying to install Signbank elsewhere, but for our own installation we can make it something else.

Having said that, that of course only postpones the problem, because our settings file is also in this repo. Thinking about this, I realize I've seen a problem like this multiple times at my other employer, and there it's always solved by creating a second `x-config` repo, like `Global-signbank-config`. This repo should be private, and as a free bonus would also be a good place to store issues related to the Lightning installation, a related problem that we've discussed before.

**Comment by vanlummelhuizen** [comment]
For Mediate I use a script that reads variables form a `.env` file: https://github.com/Mediate18/Mediate/blob/master/mediate/decouple.py. The settings.py file contains things like

https://github.com/Mediate18/Mediate/blob/53b3b5a4fc41e736761c08b26ef53d0718d6e1a3/mediate/settings.py#L24-L30

Other options:

* https://django-environ.readthedocs.io/en/latest/
* https://pypi.org/project/python-decouple/

**Comment by Woseseltops** [comment]
@Jetske , a separate config repo has been created in the mean time, but now the question is how this repo should work together with this one... could you make a proposal here? Do we need git submodules? Some sort of setup with symlinks? Something else?

**Comment by vanlummelhuizen** [comment]
@Woseseltops Do you mean by *separate config repo* this one: https://github.com/Signbank/Global-signbank-RU-instance? If so, I thought we would use this repo only for the issues where we would discuss all things concerning the RU instance. 

Perhaps we could think of something like *instance flavors* (or types, classes, categories etc.), so that we have a _RU flavor_ and an _ASL flavor_, and at the same time make something that contains instance specific things (SECRET_KEY, DATABASES, LANGUAGES etc), e.g. a .env file or environment variables. Some of the instance specific settings could be noted on our wiki or in our password manager. This setup would avoid making two repo's cooperate which is yet  another, possibly problematic, dependency.

**Comment by Jetske** [comment]
What I've done now is, like you say, made it a server specific setting that is loaded into the admin url (in urls.py). The default value is still 'admin'. For the specific instance I first tried to read the url as string from a file outside of the repo (atm in writable/adminurl.txt). I don't know if that's allowed to do in the Django settings, but it works. Then I changed it to be an environment variable, which is probably a neater solution, but you need to make sure to gitignore it. And then you don't need to link to the other repository so I think so far that is the best option.

What do you think?

**Comment by vanlummelhuizen** [comment]
Now that I reread my previous comment, especially

> something that contains instance specific things

, I remember in Signbank we have of course the `.gitignore`d file `signbank/settings/server_specific/server_specific.py` that contains instance specific settings. For Global Signbank this is for the most part a copy of `signbank/settings/server_specific/global_sb_new_aj.py` with a few changes and additions.

@Jetske Is your solution easily transfered to the `signbank/settings/server_specific/server_specific.py` setup?

**Comment by Woseseltops** [comment]
> @Woseseltops Do you mean by separate config repo this one: https://github.com/Signbank/Global-signbank-RU-instance? 

Yes!

> If so, I thought we would use this repo only for the issues where we would discuss all things concerning the RU instance.

You might have missed this comment then, which I apparently incorrectly assumed you agreed with:

>  creating a second x-config repo, like Global-signbank-config. This repo should be private, and as a free bonus would also be a good place to store issues related to the Lightning installation, a related problem that we've discussed before.

As we haven't reached consensus, let me try to convince you guys:
* The main problem we have is that we want to keep some settings out of this repository for security reasons.
* Something using `.gitignore` will obviously work, but creates the new problem that our settings are no longer under version control. I would say this is acceptable when you have only 5 settings or so, but the settings file we use on Lightning is >100 lines; indeed, I have relied on version control for this file several times.
* The proposed solution is to use a separate private repository for settings files, but this indeed has the downside of complicating the installation even further... unless we set it up in such a way that this public repo will also work _without_ extra repository, and the private repo functions as an add-on? A simple approach: `server_specific.py` is now a symlink to a file in same the public repo... it could easily be a symlink to a file in another private repo cloned to the same machine?

Let me know what you think!
